InformationScripting/src/dataformat/Tuple.cpp

```diff
@@ -24,40 +24,49 @@
 **
 ***********************************************************************************************************************/
 
-#include "InformationEdge.h"
+#include "Tuple.h"
 
 namespace InformationScripting {
 
-const QString InformationEdge::COUNT_PROPERTY_{"count"};
-const QString InformationEdge::NAME_PROPERTY_{"name"};
+Tuple::Tuple(std::initializer_list<NamedProperty> initialValue)
+	: values_{initialValue}
+{}
 
-InformationEdge::InformationEdge(InformationNode* from, InformationNode* to, const QString& name,
-											Orientation orientation)
-	: PropertyMap{{{COUNT_PROPERTY_, 1}, {NAME_PROPERTY_, name}}}, from_{from}, to_{to}, orientation_{orientation}
+Tuple::Tuple(const QList<NamedProperty>& initialValues)
+	: values_{initialValues}
 {}
 
-InformationEdge::InformationEdge(const InformationEdge& other)
-	: PropertyMap(other)
+NamedProperty Tuple::get(int index) const
+{
+	return values_.at(index);
+}
+
+QList<NamedProperty> Tuple::getAll(const QString& key)
+{
+	QList<NamedProperty> result;
+	for (auto p : values_)
+		if (p.first == key) result << p;
+	return result;
+}
+
+void Tuple::add(NamedProperty p)
 {
-	orientation_ = other.orientation_;
+	values_.append(p);
 }
 
-void InformationEdge::incrementCount()
+uint Tuple::hashValue(uint seed) const
 {
-	Property& count = (*this)[COUNT_PROPERTY_];
-	count = static_cast<int>(count) + 1;
+	return qHashRange(values_.begin(), values_.end(), seed);
 }
 
-void InformationEdge::setFrom(InformationNode* from)
+Tuple::const_iterator Tuple::find(const QString& key) const
```

Comment:
Hmm, I don't think we should have both a `find` and a `get` method. Let's have:
- only two get methods (int + string)
- only two [] operators (int + string). this might be even nicer, but I don't know.